client/packages/common/src/ui/forms/JsonForms/components/Text.tsx

```diff
@@ -109,6 +109,28 @@ const UIComponent = (props: ControlProps) => {
         })
       : zErrors ?? errors ?? customErrors;
 
+  const { core, dispatch } = useJsonForms();
+  useEffect(() => {
+    if (!core || !dispatch) {
+      return;
+    }
+    const currentErrors = core?.errors ?? [];
+    if (customErrors) {
+      dispatch(
+        Actions.updateErrors([
+          ...currentErrors,
+          {
+            instancePath: path,
+            message: customErrors,
+            schemaPath: path,
+            keyword: '',
+            params: {},
+          },
+        ])
+      );
+    }
+  }, [core, dispatch, customErrors]);
```

Comment:
If there was a custom validation and it failed you where still able to save the form